Utils.fs

```diff
@@ -280,6 +280,12 @@ let foldFastTerm (f : 'State -> 'T -> 'State option)
 
     fft f s
 
+/// Repeatedly apply f until a fixed point is reached
```

Comment:
Should perhaps pass a bound here and throw an error if it's exceeded? This seems like a place it'd be easy to accidentally introduce nontermination.